File: problems/31-bellman-ford/001-cheapest-flights-k-stops/python/solution.py
def find_cheapest_price(n: int, flights: list[list[int]], src: int, dst: int, k: int) -> int:
    inf = 10**9
    dist = [inf] * n
    dist[src] = 0

    for _ in range(k + 1):
        next_dist = dist[:]
        for source, target, price in flights:
            if dist[source] != inf:
                next_dist[target] = min(next_dist[target], dist[source] + price)
        dist = next_dist

    return -1 if dist[dst] == inf else dist[dst]

File: problems/31-bellman-ford/001-cheapest-flights-k-stops/java/Solution.java
import java.util.Arrays;

public class Solution {
    public int findCheapestPrice(int n, int[][] flights, int src, int dst, int k) {
        final int INF = 1_000_000_000;
        int[] dist = new int[n];
        Arrays.fill(dist, INF);
        dist[src] = 0;

        for (int round = 0; round <= k; round++) {
            int[] next = dist.clone();
            for (int[] flight : flights) {
                int from = flight[0];
                int to = flight[1];
                int price = flight[2];
                if (dist[from] != INF) {
                    next[to] = Math.min(next[to], dist[from] + price);
                }
            }
            dist = next;
        }

        return dist[dst] == INF ? -1 : dist[dst];
    }
}
